prov/efa/src/efa_domain.c

```diff
@@ -228,9 +230,50 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 	domain->util_domain.mr_map.mode &= ~FI_MR_PROV_KEY;
 	domain->fab = fabric;
 
+	domain->util_domain.domain_fid.mr = &efa_domain_mr_ops;
+
 	*domain_fid = &domain->util_domain.domain_fid;
 
 	if (efa_mr_cache_enable) {
+		/*
+		 * Register a temporary buffer and call ibv_fork_init() to
+		 * determine if fork support is enabled.
+		 *
+		 * libibverbs maintains a global variable to determine if any
+		 * registrations have occurred before ibv_fork_init() is called.
+		 * We can take advantage of this to determine if
+		 * ibv_fork_init() was called before Libfabric was initialized.
+		 *
+		 * As a temporary work around, EFA will return an error when
+		 * when the cache is enabled and fork support is enabled.
+		 * Using libibverbs fork support and deregistering regions
+		 * after they are unmapped has known bugs; deregistering
+		 * regions after they are unmapped is required for Libfabric
+		 * memory monitor intercept functions to work.
+		 */
+		tmp_reg_buf = malloc(ofi_get_page_size());
+		if (!tmp_reg_buf) {
+			ret = -FI_ENOMEM;
+			goto err_free_info;
+		}
+
+		ret = fi_mr_reg(*domain_fid, tmp_reg_buf, ofi_get_page_size(),
+		                FI_SEND, 0, 0, 0, &tmp_mr, NULL);
+		if (ret) {
+			ret = -FI_ENOMEM;
+			goto err_free_info;
+		}
+
+		ret = ibv_fork_init();
+
+		fi_close(&tmp_mr->fid);
+		free(tmp_reg_buf);
+
+		if (ret != EINVAL) {
+			ret = -FI_EINVAL;
```

Comment:
Do we want to print some warning message here?

Reply:
Yep, I haven't come up with an a good one yet. :)

Reply:
sorry, missed your first comment. 

We also need to update the error message in ```efa_atfork_callback``` too. Currently we advise user to set RDMAV_FOR_SAFE, which won't work.

Reply:
Ah, good point, thanks. I will update that comment too.